The job **`aarch64-gnu-llvm-21-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/26128715792/job/76848763008) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/76848763008) [(plain)](https://github.com/rust-lang/rust/commit/c879c58eb99b7e15ee1cc573328771c86fc59135/checks/76848763008/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
1 error: type `Shared` from private dependency 'shared' in public interface
-   --> $DIR/shared_both_private.rs:24:1
+   --> $DIR/shared_both_private.rs:23:1
3    |
4 LL | pub fn leaks_priv() -> shared::Shared {
5    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

11    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
12 
13 error: type `Shared` from private dependency 'shared' in public interface
---
To only update this specific test, also pass `--test-args privacy/pub-priv-dep/shared_both_private.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/privacy/pub-priv-dep/shared_both_private.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/privacy/pub-priv-dep/shared_both_private" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "incomplete_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/privacy/pub-priv-dep/shared_both_private/auxiliary" "-Zunstable-options" "--extern" "priv:shared=/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/privacy/pub-priv-dep/shared_both_private/auxiliary/libshared.so" "--extern" "priv:reexport=/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/privacy/pub-priv-dep/shared_both_private/auxiliary/libreexport.so"
stdout: none
--- stderr -------------------------------
error: type `Shared` from private dependency 'shared' in public interface
##[error]  --> /checkout/tests/ui/privacy/pub-priv-dep/shared_both_private.rs:23:1
   |
LL | pub fn leaks_priv() -> shared::Shared {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> /checkout/tests/ui/privacy/pub-priv-dep/shared_both_private.rs:18:9
   |
```

</details>

--- a/tests/ui/privacy/pub-priv-dep/shared_both_private.rs
+++ b/tests/ui/privacy/pub-priv-dep/shared_both_private.rs
@@ -1,7 +1,6 @@
 //@ aux-crate:priv:shared=shared.rs
-//@ aux-crate:reexport=reexport.rs
+//@ aux-crate:priv:reexport=reexport.rs
 //@ compile-flags: -Zunstable-options
-//@ check-pass
 
 // A shared dependency, where a private dependency reexports a public dependency.
 //
@@ -21,12 +20,12 @@
 extern crate shared;
 extern crate reexport;
 
-// FIXME: This should trigger.
 pub fn leaks_priv() -> shared::Shared {
+    //~^ ERROR type `Shared` from private dependency 'shared' in public interface
     shared::Shared
 }
 
-// FIXME: This should trigger.
 pub fn leaks_priv_reexport() -> reexport::Shared {
+    //~^ ERROR type `Shared` from private dependency 'shared' in public interface
     reexport::Shared
 }

--- a/tests/ui/privacy/pub-priv-dep/shared_both_private.stderr
+++ b/tests/ui/privacy/pub-priv-dep/shared_both_private.stderr
@@ -1,0 +1,20 @@
+error: type `Shared` from private dependency 'shared' in public interface
+  --> $DIR/shared_both_private.rs:23:1
+   |
+LL | pub fn leaks_priv() -> shared::Shared {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: the lint level is defined here
+  --> $DIR/shared_both_private.rs:18:9
+   |
+LL | #![deny(exported_private_dependencies)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: type `Shared` from private dependency 'shared' in public interface
+  --> $DIR/shared_both_private.rs:28:1
+   |
+LL | pub fn leaks_priv_reexport() -> reexport::Shared {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+error: aborting due to 2 previous errors
+

